### DISCUSSION TITLE
How to change the frequency of point source input signal?

### DISCUSSION THREAD

**Discussion by smallriver666** [SeisSol/SeisSol#816] [Q&A] [answered]
I  have try to change the T parameter of momentRate to get higher frequency input signals，but it seems get the same result, the results were not noticeably different.The file I use is [here](https://github.com/smallriver666/FrenquencyProblem).
If I want to generate a high-frequency signal, so that hundreds of wavelengths can appear in my grid interval, how should I set the grid size and the value of fm or T?
T=0.2
![image](https://user-images.githubusercontent.com/104721358/224333653-906d3c81-a454-47b4-8925-c0db303d383f.png)
T=0.1
![image](https://user-images.githubusercontent.com/104721358/224333755-e7fd3d05-af53-4692-9dda-fed7b3de05b6.png)
T0.=04
![image](https://user-images.githubusercontent.com/104721358/224335120-05494d97-84e0-4689-ab01-30bf1e0504d7.png)

fm=5
![image](https://user-images.githubusercontent.com/104721358/224335781-a695e700-9ef7-4c7a-839a-0fa53ed34bf9.png)

fm=10
![image](https://user-images.githubusercontent.com/104721358/224334851-2b539135-2e87-4656-9bcc-4bc16c5713b8.png)

**Comment by krenzland** [comment, marked as answer]
There is no good answer for that. Typically, you need X elements per wavelength. So, if you decrease the wavelength by a factor of Y, you also need to change the element size h by a factor of Y. 
For an empirical rule of how to determine X, see: Käser, Martin, Verena Hermann, and Josep de la Puente. "Quantitative accuracy analysis of the discontinuous Galerkin method for seismic wave propagation." Geophysical Journal International 173.3 (2008): 990-999.

**Reply by smallriver666** [reply]
> There is no good answer for that. Typically, you need X elements per wavelength. So, if you decrease the wavelength by a factor of Y, you also need to change the element size h by a factor of Y. For an empirical rule of how to determine X, see: Käser, Martin, Verena Hermann, and Josep de la Puente. "Quantitative accuracy analysis of the discontinuous Galerkin method for seismic wave propagation." Geophysical Journal International 173.3 (2008): 990-999.

So in theory, the point source I generated according to this [script](https://github.com/smallriver666/FrenquencyProblem/blob/main/fm10/generate_LOH_source_type50.py#:~:text=the%20moment%20rate%3A-,fm%3D10,-dt%3D0.0001) should be a ricker wavelet with a center frequency of 10Hz, which means that if I choose a reasonable grid size, I can get a result of more than a hundred wavelengths. (However, the ricker wavelet does not satisfy the condition that the integral value is 1, and his integral value is close to 0, will this have any effect?)

By the way, When you use ricker wavelet as point source, do you also use the input file format of Sourcetype type=50? Or an input file in another format?
Many Thanks in advance !

![image](https://user-images.githubusercontent.com/104721358/224376659-2a499ea0-2c23-489d-ab3a-ee1ee4432b01.png)
